tdms/tests/include/abstract_array_test_class.h

```diff
@@ -0,0 +1,204 @@
+/**
+ * @file array_test_class.h
+ * @brief Defines the abstract class for testing objects defined in arrays.h
+ */
+#pragma once
+
+#include <catch2/catch_test_macros.hpp>
+#include <spdlog/spdlog.h>
+#include <string>
+
+#include "arrays.h"
+
+/**
+ * @brief Abstract container for the tests that classes in arrays.h will have to pass.
+ *
+ * Private methods are empty, and designed to be overridden in the derived class. However this also means that definitions for these tests can be left out of those subclasses, whilst not changing the run_all_tests() function.
+ *
+ */
+class AbstractArrayTest {
+protected:
+  int empty_dimensions[2] = {0, 1};   //< For initialising empty arrays
+  int I_tot = 4, J_tot = 8, K_tot = 4;//< For mimicing simulation grid dimensions
+  int n_numeric_elements =
+          8;//< For standardising the number of elements we initialise MATLAB vectors with
+  int dimensions_2d[2] = {1, 1};               //< For defining 2D MATLAB arrays
+  int dimensions_3d[3] = {I_tot, J_tot, K_tot};//< For defining 3D MATLAB arrays
+
+  // To point to the (top-level) matlab inputs the classes require. Top-level structure ensures mxDestroyArray clears all sub-arrays too
+  mxArray *matlab_input = nullptr;
+  // Flags if we have created an mxArray manually, so we are forced to destroy to free memory when done
+  bool matlab_input_assigned = false;
+
+  /**
+   * @brief Destroys the matlab_input array if it is assigned.
+   *
+   * @returns true If mxDestroyArray was called successfully
+   * @returns false matlab_input was not assigned, so no deletion was necessary
+   */
+  bool destroy_matlab_input() {
+    if (matlab_input_assigned) {
+      mxDestroyArray(matlab_input);
+      matlab_input_assigned = false;
+      return true;
+    } else {
+      return false;
+    }
+  }
```

Comment:
I debated it, but ultimately I like having the `bool` because:
a) It forces me to actually check that I've freed MATLAB memory
b) It forces me to be aware that _some_ MATLAB functions do cleanup after themselves
c) It's a good indicator for where we're going to have to be careful with free-ing memory when #70 gets underway 
d) MATLAB pointers are mildly terrifying and the less I have to interact with them, the better